### PR TITLE
Slippage

### PR DESCRIPTION
**Pull request by supercontracts**

### CODE REVIEW

File: tests/test_upgrade.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
+from brownie import *
+import brownie
+from dotmap import DotMap
```
Comment (shuklaayush) [inline tests/test_upgrade.py:3]: Doesn't look like these are used

File: contracts/IbbtcVaultZap.sol [hunk 2]
```diff
@@ -197,11 +207,37 @@ contract IbbtcVaultZap is PausableUpgradeable {
...
+            _mintIbbtc
+        );
+
+        ERC20Upgradeable[4] memory assets = [
+            ERC20Upgradeable(address(IBBTC)),
+            ERC20Upgradeable(address(RENBTC)),
+            ERC20Upgradeable(address(WBTC)),
+            ERC20Upgradeable(address(SBTC))
+        ];
+        uint256 virtualPrice = CURVE_REN_POOL.get_virtual_price();
```
Comment (shuklaayush) [inline contracts/IbbtcVaultZap.sol:226]: Shouldn't this be Ibbtc pool? Also need to convert ibbtc to wibbtc

File: contracts/IbbtcVaultZap.sol [hunk 3]
```diff
@@ -197,11 +207,37 @@ contract IbbtcVaultZap is PausableUpgradeable {
...
+        for (uint256 i = 0; i < 4; i++) {
+            amount = amount.add(
+                depositAmounts[i].mul(1e36).div(virtualPrice).div(
+                    10**uint256(assets[i].decimals())
+                )
+            );
```
Comment (shuklaayush) [inline contracts/IbbtcVaultZap.sol:232]: Can divide by virtual price outside the loop as well to prevent rounding errors

Comment (tabshaikh) [reply contracts/IbbtcVaultZap.sol]: done

File: contracts/IbbtcVaultZap.sol (hunk 2 above)
Comment (tabshaikh) [reply contracts/IbbtcVaultZap.sol:234]: done

File: tests/test_upgrade.py (hunk 1 above)
Comment (tabshaikh) [reply tests/test_upgrade.py]: removed

File: contracts/IbbtcVaultZap.sol [hunk 4]
```diff
@@ -197,11 +211,44 @@ contract IbbtcVaultZap is PausableUpgradeable {
...
+        for (uint256 i = 0; i < 4; i++) {
+            if (depositAmounts[i] != 0) {
+                amount = amount.add(
+                    depositAmounts[i].mul(1e18).div(
+                        10**uint256(assets[i].decimals())
+                    )
+                );
+            }
+        }
+        amount = amount.mul(1e8).div(virtualPrice);
```
Comment (shuklaayush) [inline contracts/IbbtcVaultZap.sol:244]: Shouldn't this be `1e18`? Can also just multiply by `1e36` inside the loop (or multiply by `10 ** (36 - decimals)`)

**Review by shuklaayush — APPROVED** [review]
Lgtm. `expectedAmount` would output shares assuming a balanced curve pool

**Pull request merged**